homeassistant/components/switch/homematicip_cloud.py

```diff
@@ -0,0 +1,81 @@
+"""
+Support for HomematicIP switch.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/switch.homematicip_cloud/
+"""
+
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.components.homematicip_cloud import (
+    HomematicipGenericDevice, DOMAIN as HOMEMATICIP_CLOUD_DOMAIN,
+    ATTR_HOME_ID)
+
+DEPENDENCIES = ['homematicip_cloud']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_POWER_CONSUMPTION = 'power_consumption'
+ATTR_ENERGIE_COUNTER = 'energie_counter'
+ATTR_PROFILE_MODE = 'profile_mode'
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Set up the HomematicIP switch devices."""
+    from homematicip.device import (
+        PlugableSwitch, PlugableSwitchMeasuring,
+        BrandSwitchMeasuring)
+
+    if discovery_info is None:
+        return
+    home = hass.data[HOMEMATICIP_CLOUD_DOMAIN][discovery_info[ATTR_HOME_ID]]
+    devices = []
+    for device in home.devices:
+        if isinstance(device, BrandSwitchMeasuring):
```

Comment:
If we're not using this case, don't add it.

Reply:
Hi Martin, BrandSwitchMeasuring inherits the classes PlugableSwitchMeasuring, so this device would also be added as a switch. But I want to have this device instantiated in the light platform.

Reply:
Ok, but then add a second condition on the case below instead.
```py
if (isinstance(device, PlugableSwitchMeasuring) and
        not isinstance(device, BrandSwitchMeasuring)):
    devices.append(...)
```

Reply:
Or keep this case and add a comment why it's there. I guess it's less code to keep this case, since you guard both cases below.